feat: conditionally enable plugins for multi-app environments (#130)
coreruleset/documentation: content/concepts/plugins.md

Alternatively, it is also valid to disable a plugin by renaming a plugin file from `plugin-before.conf` to `plugin-before.conf.disabled`.

## Conditionally enable plugins for multi-application environments

> If CRS is installed on a reverse-proxy or a web server with multiple web applications, then you may wish to only enable certain plugins (such as rule exclusion plugins) for certain VirtualHosts. This ensures that rules designed for a specific web application are only enabled for the intended web application, reducing the scope of any possible bypasses within a plugin.

Reviewer: ```suggestion
If CRS is installed on a reverse-proxy or a web server with multiple web applications, then you may wish to only enable certain plugins (such as rule exclusion plugins) for certain `VirtualHosts`. This ensures that rules designed for a specific web application are only enabled for the intended web application, reducing the scope of any possible bypasses within a plugin.
```

> If CRS is installed on a reverse-proxy or a web server with multiple web applications, then you may wish to only enable certain plugins (such as rule exclusion plugins) for certain `VirtualHosts`. This ensures that rules designed for a specific web application are only enabled for the intended web application, reducing the scope of any possible bypasses within a plugin.

Reviewer: ```suggestion
If CRS is installed on a reverse-proxy or a web server with multiple web applications, then you may wish to only enable certain plugins (such as rule exclusion plugins) for certain virtual hosts (`VirtualHost` Apache httpd, `location` in nginx). This ensures that rules designed for a specific web application are only enabled for the intended web application, reducing the scope of any possible bypasses within a plugin.
```

Author: [Location](https://nginx.org/en/docs/http/ngx_http_core_module.html#location) directive in Nginx is equivalent to [Apache's Directory](https://httpd.apache.org/docs/2.4/mod/core.html#directory) directive, both respectively live within the Server Block and Virtual Host context.

Reviewer: Right. So `Server` and `VirtualHost` then.

> Most plugins provide an example to disable the plugin in the file `plugin-config.conf`, you can define the `WebAppID` variable for each `VirtualHost` and then disable the plugin when the `WebAppID` variable doesn't match.

Reviewer: ```suggestion
Most plugins provide an example to disable the plugin in the file `plugin-config.conf`, you can define the `WebAppID` variable for each virtual host and then disable the plugin when the `WebAppID` variable doesn't match.
```

> Most plugins provide an example to disable the plugin in the file `plugin-config.conf`, you can define the WebAppID variable for each virtualhosts and then disable the plugin if the WebAppID variable doesn't equal, for example WordPress.

Reviewer: ```suggestion
Most plugins provide an example to disable the plugin in the file `plugin-config.conf`, you can define the `WebAppID` variable for each `VirtualHost` and then disable the plugin when the `WebAppID` variable doesn't match.
```

See: https://github.com/owasp-modsecurity/ModSecurity/wiki/Reference-Manual-(v2.x)#secwebappid

> Below is an example for enabling only the WordPress plugin for WordPress `VirtualHost`s:

Reviewer: ```suggestion
Below is an example for enabling only the WordPress plugin for WordPress virtual hosts:
```

> Below is an example for only enabling the WordPress plugin for WordPress VirtualHosts:

Reviewer: ```suggestion
Below is an example for enabling only the WordPress plugin for WordPress `VirtualHost`s:
```

```
SecRule &TX:wordpress-rule-exclusions-plugin_enabled "@eq 0" \
    "id:9507010,\
    phase:1,\
    pass,\
    nolog,\
    ver:'wordpress-rule-exclusions-plugin/1.0.0',\
    chain"
    SecRule WebAppID "!@streq wordpress" \
        "t:none,\
        setvar:'tx.wordpress-rule-exclusions-plugin_enabled=0'"
```

> ⚠️ Warning: As of 05/06/2024, Coraza doesn't support the use of WebAppID, you can use the host header instead of the WebAppID variable:

Reviewer: ```suggestion
⚠️ Warning: As of 05/06/2024, Coraza doesn't support the use of WebAppID, you can use the`Host` header instead of the `WebAppID` variable:
```
